eclair-core/src/main/scala/fr/acinq/eclair/payment/Bolt11Invoice.scala

```diff
@@ -355,24 +364,16 @@ object PaymentRequest {
   /**
    * Features supported or required for receiving this payment.
    */
-  case class PaymentRequestFeatures(bitmask: BitVector) extends TaggedField {
-    lazy val features: Features = Features(bitmask)
-    lazy val allowMultiPart: Boolean = features.hasFeature(Features.BasicMultiPartPayment)
-    lazy val allowPaymentSecret: Boolean = features.hasFeature(Features.PaymentSecret)
-    lazy val requirePaymentSecret: Boolean = features.hasFeature(Features.PaymentSecret, Some(FeatureSupport.Mandatory))
-    lazy val allowTrampoline: Boolean = features.hasFeature(Features.TrampolinePayment)
-
-    def toByteVector: ByteVector = features.toByteVector
+  case class InvoiceFeatures(features: Features) extends TaggedField
 
-    def areSupported(nodeParams: NodeParams): Boolean = nodeParams.features.areSupported(features)
-
-    override def toString: String = s"Features(${bitmask.toBin})"
-  }
-
-  object PaymentRequestFeatures {
-    def apply(features: Map[Feature with InvoiceFeature, FeatureSupport]): PaymentRequestFeatures = PaymentRequestFeatures(long2bits(features.foldLeft(0L) {
-      case (current, (feature, support)) => current + (1L << feature.supportBit(support))
-    }))
+  def featuresBitmask(features: Features): BitVector = {
+    val bits = features.toByteVector.bits
+    val highest = bits.toIndexedSeq.indexOf(true)
+    val nonPadded = if (highest == -1) BitVector.empty else bits.drop(highest)
+    nonPadded.size % 5 match {
+      case 0 => nonPadded
+      case remaining => BitVector.fill(5 - remaining)(high = false) ++ nonPadded
+    }
```

Comment:
Since this borrows heavily from `long2bits`, what about refactoring the common parts?
I've done it [here](https://github.com/ACINQ/eclair/commit/beb604a394b06553c5ea94dd7c71edaeab3e358b) if you like it.

Reply:
Done